File: API/AcQua TaQuot/Entities.Mapper/Commanditaire/Commanditaire.cs
using Entities.Models;
using Shared.DataTransfertObject;

namespace Entities.Mapper
{
    public static partial class CommanditaireMapper
    {
        public static CommanditaireDTO ToCommanditaireDTO(this Commanditaire entity)
        {
            return new CommanditaireDTO(entity.Id, entity.Nom);
        }
    }
}

File: APP/TaQuot/src/models/Commanditaire/Commanditaire.ts
export default class CommanditaireDTO{
    id = '';
    nom = ''
}

export class Commanditaire extends CommanditaireDTO{

    public constructor(dto : CommanditaireDTO | null = null){
        super();

        if(dto?.id){
            this.id = dto.id;
        }
        
        if(dto?.nom){
            this.nom = dto.nom;
        }
    }

}